src/stan/io/stan_csv_reader.hpp

```diff
@@ -194,6 +194,11 @@ namespace stan {
         }
         ss.seekg(std::ios_base::beg);
 
+        // All the code below is specific to NUTS
+        // which has at least 4 lines of adaptation output
+        if (lines < 4)
```

Comment:
@akucukelbir, this needs a test. I didn't see a change to the `stan_csv_reader_test.cpp` and I didn't see an appropriate test anywhere else, but could have missed it.